astro-run docs (#1532)
astronomer/docs: astro/test-and-troubleshoot-locally.md

As you develop data pipelines on Astro, Astronomer recommends running and testing your DAGs locally before deploying your project to a Deployment on Astro. This document provides information about testing and troubleshooting DAGs in a local Apache Airflow environment with the Astro CLI.

> ## Setup your Local Testing Environment

Reviewer: ```suggestion
## Set up your local testing environment
```

> To test your code, the first step is to setup your local testing environment by [creating an Astro project](create-project.md).

Reviewer: ```suggestion
To test your code, [create an Astro project](create-project.md).
```

> ## Run and Debug DAGs with Astro Run

Reviewer: ```suggestion
## Run and debug DAGs with astro run
```

> Use the `astro run` command to compile and run a DAG from the command line . When you run this command, your DAGs run within a Docker container built from your local `Dockerfile` that contains your DAGs, DAG utility files, python requirements, and environment variables from the `.env` file. This command lets you develop DAGs without using the Airflow Webserver.

Reviewer 2: ```suggestion
The `astro run` command is the easiest and fastest way to test DAGs from the command line. You can run an individual DAG and see task successes or failure in your terminal without having to manually go to the Airflow UI and view logs.

When you run this command, your DAGs run in a Docker container that is built from your local `Dockerfile` and contains your DAG utility files, Python requirements, and environment variables from the `.env` file. It does not need the Airflow webserver or scheduler to run.
```

> You can compile an run a DAG with only the command line using the `astro run` command. This command will run your DAGs within a Docker container bulit from your local `Dockerfile` containing your DAGs, DAG utility files, python requirments, and env vars from your `.env` file. This is helpful for users who do not wish to use the Airflow Webserver while developing a DAG.

Reviewer: ```suggestion
Use the `astro run` command to compile and run a DAG from the command line . When you run this command, your DAGs run within a Docker container built from your local `Dockerfile` that contains your DAGs, DAG utility files, python requirements, and environment variables from the `.env` file. This command lets you develop DAGs without using the Airflow Webserver.
```

> To quickly run a DAG located within your local `/dags` directory run:

Reviewer: ```suggestion
To run a DAG located within your local `/dags` directory run:
```

```sh
astro run <dag-id>
```

> This command will run all the tasks in your DAG sequentally. Any errors produced by your code while parsing or running your DAG will show up in the command line. You can provide variables and connections to your DAGs using the [Airflow Settings file](develop-project.md#configure-airflow_settingsyaml-local-development-only). The command will use `airflow_settings.yaml` by default but you can specify any yaml file you'd like with the `--airflow-settings` flag.

Reviewer: ```suggestion
All the tasks in your DAG run sequentially. Any errors produced by your code while parsing or running your DAG appear in the command line. To provide variables and connections to your DAGs, use the [Airflow Settings file](develop-project.md#configure-airflow_settingsyaml-local-development-only). By default, the command uses `airflow_settings.yaml` , but you can use the `--airflow-settings` flag to specify a specific yaml file.
```

> All the tasks in your DAG run sequentially. Any errors produced by your code while parsing or running your DAG appear in the command line. To provide variables and connections to your DAGs, use the [Airflow Settings file](develop-project.md#configure-airflow_settingsyaml-local-development-only). By default, the `astro run` command uses `airflow_settings.yaml`, but you can use the `--airflow-settings` flag to specify a different yaml file.
> 
> To use variables and connections stored in your local postgres instance, run `astro run` while a local instance of Airflow is running. To run your project in a local Airflow environment, see [Build and run a project](develop-project.md#build-and-run-a-project-locally).

Reviewer 2: ```suggestion
All the tasks in your DAG run sequentially. Any errors produced by your code while parsing or running your DAG appear in the command line.

To provide variables and connections to your DAGs with the command, use the [`airflow_settings.yaml` file](develop-project.md#configure-airflow_settingsyaml-local-development-only) or use the `--airflow_settings` flag to specify a different yaml file. To use variables and connections that are already set in the Airflow UI of a local Airflow environment, run `astro run` while your local Airflow environment is running. See [Build and run a project](develop-project.md#build-and-run-a-project-locally).
```

> If you'd like to utilize variables and connections stored in your local postgres instance run `astro run` while a local instance of Airflow is running. To run your project in a local Airflow environment, see [Build and run a project](develop-project.md#build-and-run-a-project-locally).

Reviewer: ```suggestion
To use variables and connections stored in your local postgres instance, run `astro run` while a local instance of Airflow is running. To run your project in a local Airflow environment, see [Build and run a project](develop-project.md#build-and-run-a-project-locally).
```

Use the `astro run` command to compile and run a DAG from the command line. When you run this command, your DAGs run within a Docker container built from your local `Dockerfile` that contains your DAGs, DAG utility files, python requirements, and environment variables from the `.env` file. This command lets you develop DAGs without using the Airflow Webserver.

To run a DAG located within your local `/dags` directory run:

```sh
astro run <dag-id>
```

All the tasks in your DAG run sequentially. Any errors produced by your code while parsing or running your DAG appear in the command line. To provide variables and connections to your DAGs, use the [Airflow Settings file](develop-project.md#configure-airflow_settingsyaml-local-development-only). By default, the `astro run` command uses `airflow_settings.yaml`, but you can use the `--airflow-settings` flag to specify a different yaml file.

To use variables and connections stored in your local postgres instance, run `astro run` while a local instance of Airflow is running. To run your project in a local Airflow environment, see [Build and run a project](develop-project.md#build-and-run-a-project-locally).

> ## Test DAGs with the Astro CLI

Reviewer 2: @sunkickr Would it make sense to just add this section here?

`astro dev pytest` runs this default test alongside any other custom tests that you add to the `tests` directory. For more information about this command, see the [CLI command reference](cli/astro-dev-pytest.md).

### Run a DAG with `astro run`

> The `astro run` command is the easiest and fastest way to test DAGs from the command line. You can run an individual DAG and see task statuses in your terminal without using the Airflow UI.

Reviewer: ```suggestion
Use the `astro run` command to test DAGs from the command line. You can run an individual DAG and see task statuses in your terminal without using the Airflow UI.
```
![image](https://user-images.githubusercontent.com/104205257/203584464-1c6ff3a0-3956-4a84-a818-70cc20aabcd0.png)

> Use the `astro run` command to test DAGs from the command line. You can run an individual DAG and see task statuses in your terminal without using the Airflow UI.
> 
> When you run this command, your DAGs run in a Docker container that is built from your local `Dockerfile` and it contains your DAG utility files, Python requirements, and environment variables from the `.env` file. It does not need the Airflow webserver or scheduler to run.

Reviewer 2: ```suggestion
Use the `astro run` command to test a DAG from the command line. When you run the command, the CLI compiles your DAG and runs it in a single Airflow worker container based on your Astro project configurations, including your `Dockerfile`, DAG utility files, Python requirements, and environment variables. You can see task logs and task success or failure directly in your terminal and without needing to go to the Airflow UI. You can only run one DAG at a time.

This command is an alternative to running `astro dev restart` every time you make a change to your DAG. Running DAGs without a scheduler or webserver improves the speed at which you can develop and test data pipelines.
```

To run a DAG located within your local `/dags` directory run:

```sh
astro run <dag-id>
```

> All the tasks in your DAG run sequentially. Any errors produced by your code while parsing or running your DAG appear in the command line. For more information about this command, see the [CLI command reference](cli/astro-run.md)/

Reviewer 2: ```suggestion
All the tasks in your DAG run sequentially. Any errors produced by your code while parsing or running your DAG appear in the command line. For more information about this command, see the [CLI command reference](cli/astro-run.md).
```
